File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    let answer = [];
    const scoreMap = new Map();
    
    for(let i=0; i < name.length; i++){
        scoreMap.set(name[i], yearning[i]);
    }
    
    for(let i=0; i < photo.length; i++){
        let sum = 0;
        for(let j=0; j< photo[i].length; j++){
            if(scoreMap.get(photo[i][j])) {
                sum += scoreMap.get(photo[i][j]);
            }
        }
        answer.push(sum);
    }
    
    
    return answer;
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        Map<String, Integer> nameScore = new HashMap<String,Integer>();
        
        for(int i = 0; i < name.length; i++){
            nameScore.put(name[i], yearning[i]);
        }
        
        for(int i = 0; i < photo.length; i++){
            int sum = 0;
            for(int j = 0; j < photo[i].length; j++){
                if(nameScore.get(photo[i][j]) != null){
                    sum += nameScore.get(photo[i][j]);
                }
            }
            answer[i] = sum;
        }
        
        
        return answer;
    }
}